rednotebook: 2.8 -> 2.11.1

r-ryantm opened the pull request:
Semi-automatic update generated by https://github.com/ryantm/nixpkgs-update tools. This update was made based on information from https://repology.org/metapackage/rednotebook/versions.



meta.description for rednotebook is: '"A modern journal that includes a calendar navigation, customizable templates, export functionality and word clouds"'.


meta.homepage for rednotebook is: '"http://rednotebook.sourceforge.net/"

[Release on GitHub](https://github.com/jendrikseipp/rednotebook/releases/tag/v2.11.1)


[Compare changes on GitHub](https://github.com/jendrikseipp/rednotebook/compare/v2.8...v2.11.1)

<details>
<summary>
Checks done (click to expand)
</summary>

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- /bin/sh passed the binary check.
- 1 of 1 passed binary check by having a zero exit code.
- 0 of 1 passed binary check by having the new version present in output.
- found 2.11.1 with grep in /nix/store/yisdgnnpdz3lzh3611v7w2g8mygv1xmm-rednotebook-2.11.1
- directory tree listing: https://gist.github.com/aa2dda0804b2bf496bbdd4ce2159b560
- du listing: https://gist.github.com/f2bce9733f23ba9ccb84d30a411912b0

</details>
<details>
<summary>
Rebuild report (if merged into master) (click to expand)
</summary>

4 total rebuild path(s)

1 package rebuild(s)

1 x86_64-linux rebuild(s)
1 i686-linux rebuild(s)
1 x86_64-darwin rebuild(s)
1 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
rednotebook

</details>

<details>
<summary>
Instructions to test this update (click to expand)
</summary>

Either download from Cachix:
```
nix-store -r /nix/store/yisdgnnpdz3lzh3611v7w2g8mygv1xmm-rednotebook-2.11.1 \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)

Or, build yourself:
```
nix-build -A rednotebook https://github.com/r-ryantm/nixpkgs/archive/63445c00d415f6b87a37e03928d98253e965076b.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/yisdgnnpdz3lzh3611v7w2g8mygv1xmm-rednotebook-2.11.1
ls -la /nix/store/yisdgnnpdz3lzh3611v7w2g8mygv1xmm-rednotebook-2.11.1/bin
```


</details>
<br/>


cc @orivej for testing.

r-ryantm committed:
rednotebook: 2.8 -> 2.11.1

Semi-automatic update generated by
https://github.com/ryantm/nixpkgs-update tools. This update was made
based on information from
https://repology.org/metapackage/rednotebook/versions

matthiasbeyer reviewed (approved):
Diff looks good to me.

mmahut commented:
It's missing the ``journal`` module upon execution.

```
Traceback (most recent call last):
  File "/nix/store/wp06mrdzjbmiyxvz5m2rrsmf6h8zapv4-rednotebook-2.11.1/bin/.rednotebook-wrapped", line 7, in <module>
    import journal
ModuleNotFoundError: No module named 'journal'

```

orivej committed:
rednotebook: fix "Namespace Gtk not available" error

gobject-introspection setup hook looks at the packages at $hostOffset to
populate GI_TYPELIB_PATH. We need gtk3 to be there.

orivej commented:
The import journal error is expected as it comes from this:
```python
try:
    # Load module from same directory
    import journal
    print('Starting RedNotebook from the source directory')
    journal.main()
except ImportError:
    # Load installed module
    import rednotebook.journal
    rednotebook.journal.main()
```

The actual error (also present in the current version 2.8) is:
```
Traceback (most recent call last):
  File "rednotebook-2.8/bin/.rednotebook-wrapped", line 12, in <module>
    import rednotebook.journal
  File "rednotebook-2.8/lib/python3.7/site-packages/rednotebook/journal.py", line 43, in <module>
    gi.require_version("Gtk", "3.0")
  File "python3.7-pygobject-3.32.1/lib/python3.7/site-packages/gi/__init__.py", line 129, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

It happens because Gtk was not added to `GI_TYPELIB_PATH`. gobject-introspection did not add it because it only looks at the packages at the [hostOffset](https://github.com/NixOS/nixpkgs/blob/19.03/pkgs/development/libraries/gobject-introspection/setup-hook.sh#L13). (It had looked at gtk3 before, but cross-compilation support must have been improved.) I have fixed this by moving gobject-introspection  from `nativeBuildInputs` to `buildInputs` in the additional commit to this PR: https://github.com/NixOS/nixpkgs/pull/64189/commits/066c08b764415c264fb6e94b190116384035bb7f .

orivej commented:
@jtojnar, my fix will break if `make_gobject_introspection_find_gir_files` is moved to `$targetOffset` (https://github.com/NixOS/nixpkgs/issues/56943#issuecomment-472568643). Should I list `gobject-introspection` in both `nativeBuildInputs` and `buildInputs`, or just disable `strictDeps`?

orivej commented:
I have decided to disable `strictDeps` until `gobject-introspection` in `nativeBuildInputs` is supported .
